### PR TITLE
Refactor/count based retention

### PR DESCRIPTION
**Pull request by ndcampbell**

### CODE REVIEW

File: src/lavatory/utils/artifactory.py [hunk 1]
```diff
@@ -153,31 +157,72 @@ def retain(self, spec_project, depth=3, terms=None, count=None, weeks=None):
...
             if weeks:
                 now = datetime.datetime.now()
                 before = now - datetime.timedelta(weeks=weeks)
                 created = before.strftime("%Y-%m-%dT%H:%M:%SZ")
-                filtered = self.filter(depth=depth + 1, terms=[{"path": path}, {"created": created}])
-                for artifact in filtered:
-                    purgable.append("{}/{}".format(artifact["path"], artifact["name"]))
 
+                purgable_artifacts.extend(
+                    self.filter(offset=count, depth=depth + 1, terms=[{
```
Comment (bazimov) [inline src/lavatory/utils/artifactory.py]: count variable is undefined.

File: src/lavatory/utils/artifactory.py [hunk 2]
```diff
@@ -93,30 +93,33 @@ def purge(self, dry_run, artifacts):
...
         XXX: this looks at the project level, but actually need to iterate lower at project level
         XXX: almost certainly needs to set depth parameter to get to specific build
 
-        Args:
-            terms (list): AQL terms.
-            depth (int): Depth level.
-
-        Returns:
-            results: Results from aql search.
+        Keyword arguments:
```
[the remark this answers is not in the document]
Comment (ndcampbell) [reply src/lavatory/utils/artifactory.py]: Yeah, i'll fix this. left over docs from Alex

File: src/lavatory/utils/artifactory.py [hunk 3]
```diff
@@ -93,30 +93,33 @@ def purge(self, dry_run, artifacts):
...
         for artifact in artifacts:
-            LOG.info("%s purge %s:%s", (mode, self.repo_name, artifact))
+            artifact_path = "{}/{}".format(artifact['path'], artifact['name'])
+            LOG.info("  {} purge {}:{}".format(mode, self.repo_name, artifact_path))
             if dry_run:
                 purged += 1
             else:
                 try:
-                    self.artifactory.request(artifact, method='delete')
+                    self.artifactory.request(artifact_path, method='delete')
```
[the remark this answers is not in the document]
Comment (ndcampbell) [reply src/lavatory/utils/artifactory.py]: yeah! I thought that seemed odd but left it. I'll updated

File: src/lavatory/utils/artifactory.py (hunk 1 above)
Comment (ndcampbell) [reply src/lavatory/utils/artifactory.py]: I'll fix this. This whole `retain` function I think will be removed as we refactor things out

**Comment by ndcampbell** [comment]
@bazimov fixed all your feedback. I ran yapf as well

**Pull request merged**